testing/suite.ts

```diff
@@ -0,0 +1,153 @@
+import { Hook, IHookHandler, IRunContext, Runner, test, Test } from "df/testing";
+
+export interface ISuiteOptions {
+  name: string;
+  parallel?: boolean;
+}
+
+export interface ISuiteContext {
+  suite: typeof suite;
+  test: typeof test;
+  beforeEach: IHookHandler;
+  afterEach: IHookHandler;
+  setUp: IHookHandler;
```

Comment:
i think we should be consistent with the naming of these hook handlers. I propose: `before`, `after`, `beforeAll`, `afterAll`

Reply:
`before, after, beforeEach, afterEach` as it's the same as mocha. `beforeAll` is confusing IMO, as it can be kind of interpreted either way